lang_python.adoc

```diff
@@ -33,68 +33,94 @@ Runtime library API version must match compiler version that you're using
 (i.e. if you use ksc v0.8, then you must use compatible v0.8 runtime).
 
 [[add-runtime-release]]
-==== Using release version
+==== Using release version (NOT recommended)
 
 With stable version of Kaitai Struct compiler, one can use use following
 methods:
 
 * On Debian / Ubuntu, one can install it from apt repositories:
 
-** For Python 2.x:
+** For Python 3.x:
 +
 [source,shell]
-sudo apt-get install python-kaitaistruct
+sudo apt-get install python3-kaitaistruct
 +
-Reference: https://packages.debian.org/buster/python-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python-kaitaistruct[package info for Ubuntu]
+Reference: https://packages.debian.org/buster/python3-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python3-kaitaistruct[package info for Ubuntu]
 
-** For Python 3.x:
+** For Python 2.x (DEPRECATED, python 2 is EOL since 2020-01-01):
 +
 [source,shell]
-sudo apt-get install python3-kaitaistruct
+sudo apt-get install python-kaitaistruct
 +
-Reference: https://packages.debian.org/buster/python3-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python3-kaitaistruct[package info for Ubuntu]
+Reference: https://packages.debian.org/buster/python-kaitaistruct[package info for Debian], https://packages.ubuntu.com/eoan/python-kaitaistruct[package info for Ubuntu]
+
 
-* If your project is using *requirements.txt* to manage your
-  dependencies, add the following line to it:
+* Add the following dependency specifier into
+`install_requires` of `options` section
+of your `setup.cfg` (or the arg of `setuptools.setup`
+with the same name in your `setup.py`) :
 +
 ----
 kaitaistruct
 ----
 +
-and then run `pip install -r requirements.txt` to update all your
-dependencies.
+and then run `pip install --upgrade -e ./` in the dir with the `setup.py`
+in order to install your package and all its dependencies in
+an editable way.
 
-* Otherwise, you can just install the package manually using *pip*
+* Otherwise, you can just install the package manually using `pip`
   (Python package manager):
 +
 [source,shell]
-pip install kaitaistruct
+pip install --upgrade kaitaistruct
 
 [[add-runtime-snapshot]]
-==== Using latest (snapshot) version
+==== Using latest (snapshot) version (RECOMMENDED)
 
 If you're using latest (unreleased AKA "unstable" AKA "snapshot") build
 of Kaitai Struct compiler, that will require latest runtime libraries as
 well:
 
-* If your project is using *requirements.txt* to manage your
-  dependencies, add the following line to it:
+* You can a make `pip` to use a bleeding edge version of the runtime by
+  adding a https://www.python.org/dev/peps/pep-0508/[PEP 508]
+  dependency specifier into `install_requires`:
```

Comment:
`pip` is the de-facto standard. Other package managers like `conda`, `apt`, `dnf` are not native to python ecosystem and don't use PEP 508.

Reply:
There are lots of other build tools that are native to the Python ecosystem, and some of them (e. g. [flit](https://github.com/takluyver/flit)) even use PEP 508 dependency specifiers.

It's not just about `pip` though - IMHO the entire sentence goes into too much detail that isn't really relevant here. I think it would be enough to say "You can use a bleeding edge runtime by adding this dependency to your `install_requires`: ..." without talking about pip and PEP 508.

Reply:
>There are lots of other build tools that are native to the Python ecosystem

They are not native, most of packages use setuptools, and I hope they will never be native in the sense 1/4 of packages using flit, 1/4 using poetry shit, 1/4 using setuptools and 1/4 using something else. I already have issues when some mad maintainers have chosen to use `poetry`, which has an incompatible metadata format and has a principial position to use shitty `venv`s I don't want to touch anyhow because they are damn slow, eat hdd space and cause mess in managing the packages installed in them. Standardizing `pyproject.toml` without standardizing common metadata and build recipy format was a major mistake that will haunt us for a long time.

Reply:
> They are not native

By what definition? All of these build tools are compatbile with the pyproject.toml `[build-system]` section as specified in [PEP 517](https://www.python.org/dev/peps/pep-0517/) and [PEP 518](https://www.python.org/dev/peps/pep-0518/), so they can be automatically called by other build tools to produce sdists and wheels with standard metadata.

It's not really relevant whether these build tools are compatible with setuptools. The entire point of pyproject.toml is to move away from setup.py (which is based on distutils/setuptools) as the standard build entry point, so that new build can be developed that don't rely on setuptools.

Reply:
>By what definition?

By definition that I still can install most of packages without installing a zoo of build systems using only setuptools.